test/lib/fhirValidatorTest.ts

```diff
@@ -629,68 +629,88 @@ describe('fhir validator', () => {
       try {
         // @ts-ignore
-        fhirValidator.validate([]);
-        done();
+        await fhirValidator.validate([]);
       } catch (err) {
         expect(err).to.not.be.null;
         expect(err instanceof Error).to.be.true;
         expect(Array.isArray(err.errors)).to.be.true;
         expect(err.toString()).to.contain('Resource needs to be an object.');
-        done();
       }
     });
 
-    it('fails to validate with an explicit message if a parameter without a resourceType property is provided', done => {
+    it('fails to validate with an explicit message if a parameter without a resourceType property is provided', async () => {
       try {
-        fhirValidator.validate({
+        await fhirValidator.validate({
           // @ts-ignore
           uselessProperty: 'not a resource type',
         });
-        done();
       } catch (err) {
         expect(err).to.not.be.null;
         expect(err instanceof Error).to.be.true;
         expect(Array.isArray(err.errors)).to.be.true;
         expect(err.toString()).to.contain('Resource object does not have a resource type.');
         expect(err.toString()).to.not.contain('Did you mean to submit the .fhirResource property?');
-        done();
       }
     });
 
-    it('fails to validate with an explicit message if a parameter without a resourceType property is provided but a fhirResource property exists on the object', done => {
+    it('fails to validate with an explicit message if a parameter without a resourceType property is provided but a fhirResource property exists on the object', async () => {
       try {
-        fhirValidator.validate({
+        await fhirValidator.validate({
           // @ts-ignore
           uselessProperty: 'not a resource type',
           fhirResource: {
             why: 'You might want to submit me though',
           },
         });
-        done();
       } catch (err) {
         expect(err).to.not.be.null;
         expect(err instanceof Error).to.be.true;
         expect(Array.isArray(err.errors)).to.be.true;
         expect(err.toString()).to.contain('Resource object does not have a resource type.');
         expect(err.toString()).to.contain('Did you mean to submit the .fhirResource property?');
-        done();
+      }
+    });
+
+    it('fails to validate with an explicit message if a resource contains a resource that contains another resource', async () => {
+      try {
+        await fhirValidator.validate({
+          resourceType: 'DocumentReference',
+          id: 'bazinga',
+          contained: [
+            {
+              resourceType: 'DocumentReference',
+              id: 'bazonga',
+              contained: [
+                {
+                  resourceType: 'Patient',
+                  id: 'badabibadbing',
+                },
+              ],
+            },
+          ],
+        });
+      } catch (err) {
```

Comment:
Won't this pass (incorrectly) if **no** error is thrown? In that case, we never enter the `catch` clause and the test runs to completion without executing any `expect`-statements.

Reply:
Good catch Morten. That's what I meant in my previous comment ;)